src/routes/leaveGame.ts

```diff
@@ -97,4 +67,75 @@ router.get(
   }
 )
 
+async function getGameId(
+  playerToken: string,
+  client: Client
+): Promise<string | null> {
+  const query = 'SELECT game_id FROM Players WHERE token=$1'
+  const result = await client.query(query, [playerToken])
+  return result.rowCount === 0 ? null : result.rows[0].game_id
+}
+
+async function getGameMaster(gameId: string, client: Client): Promise<string> {
+  const query = 'SELECT game_master FROM Games WHERE game_id=$1'
+  return (await client.query(query, [gameId])).rows[0].game_master
+}
+
+async function deleteGame(gameId: string, client: Client) {
+  const query = 'DELETE FROM Games WHERE game_id=$1'
+  await client.query(query, [gameId])
+}
+
+async function changeGameMaster(
+  gameId: string,
+  newGameMaster: string,
+  client: Client
+) {
+  const query = 'UPDATE Games SET game_master=$1 WHERE game_id=$2'
+  const values = [newGameMaster, gameId]
+  await client.query(query, values)
+}
+
+// This function will handle database side of handling change of game master
+// and return new game master token if there are other players, otherwise
+// return initial game master token.
+async function handleGameMasterChange(
```

Comment:
Maybe function called handelGameMasterChange should not delete game and player? It violates principle of least surprise, I think this logic should be outside this function.

Reply:
We could maybe just name it something like `handleGameMasterLeft()` and then it makes more sense

Reply:
I agree with @Kwasow as it will be better to do the delete query inside the function and it doesn't seem good to make a fuction that simpy calls two other functions.

Reply:
Yea, a name change would resolve this

Reply:
Done